tinygrad/codegen/linearizer.py

```diff
@@ -148,31 +152,34 @@ def global_store(self, i:int, idxs:List[Node], store:List[UOp]) -> List[UOp]:
         amt = len(grouped)
         idx, valid = self.sts[i].expr_idxs(k)
         assert idx == ((idx//amt)*amt), "float4 stores are always aligned"
-        store_offset_new[k] = self.uops.add(UOps.CAST, buf.dtype.vec(amt), tuple(grouped))
+        store_offset_new[k] = self.uops.add(UOps.CAST, buf.dtype.vec(amt), tuple(grouped), insert_before=insert_before)
       store_offset = store_offset_new
 
     stores = []
     for _idx, var in store_offset.items():
       idx, valid = self.sts[i].expr_idxs(_idx)
       if isinstance(buf.dtype, ImageDType):
         image_idx, valid = to_image_idx(buf.dtype.shape, idx, valid)
-        rendered_idx = self.uops.add(UOps.CAST, dtypes.int.vec(2), tuple(x.render(self.render_ops, self) for x in image_idx))
+        rendered_idx = rendered_idx = self.uops.add(UOps.CAST, dtypes.int.vec(2), \
```

Comment:
why are you assigning rendered_idx twice?

Reply:
hmm it's not github issue. might need to delete the old branch before fetching again or reset hard with the upstream one

Reply:
huh, you are correct - sorry about that. Landed fix to master.